browser/js/views/lecture/lecture.controller.js

```diff
@@ -47,12 +47,18 @@ function CreateLeactureInstance($scope, $uibModalInstance, $uibModal, LectureFac
 
   $scope.submitLecture = function() {
 
-    LectureFactory.create($scope.lectureName).then(function(lecture) {
-      $scope.createdLecture = lecture;
-    })
-    .then(function(){
-        $uibModalInstance.close();
-    })
+    if ($scope.lectureEmail) {
+      LectureFactory.create({
+        name: $scope.lectureName,
```

Comment:
Could simplify. In the template instead of `ng-model="lectureName"` you could have `ng-model="lecture.name"` (and similarly elsewhere `ng-model="lecture.lecturer"`) which means you could then just `LectureFactory.create($scope.lecture)`. Not major.